### ISSUE TITLE
This expression is not callable [3.2.0]

### ISSUE THREAD

**Issue by datapain** [vultix/ts-results#24]
Hello, I have the error after updating to the 3.2.0 version.

The error is 
```
This expression is not callable.
  Each member of the union type '(<T2, E2>(op: unknown) => Err<Error>) | (<T2, E2>(mapper: (val: boolean) => Result<T2, E2>) => Result<T2, E2>)' has signatures, but none of those signatures are compatible with each other.
getStatus(true).andThen((result) => Ok(result));
```

How to reproduce:
```
import { Result, Ok, Err } from 'ts-results';

const getStatus = (payload: boolean): Result<boolean, Error> => {
  if (payload) {
    return Ok(payload);
  }
  return Err(new Error('Payload is false'));
};

getStatus(true).andThen((result) => Ok(result));
```

If I am doing something wrong, please point it out.

typescript: 4.1.5

Thank you

also, when I tried to use typescript 4.2.0 this error is not reproduced.

But.. let's get more code

```
import { Result, Ok, Err } from 'ts-results';

const getStatus = (payload: boolean): Result<boolean, Error> => {
  if (payload) {
    return Ok(payload);
  }
  return Err(new Error('Payload is false'));
};

const processStatus = (): Result<boolean, Error> => {
  return getStatus(true)
    .andThen((result) => Ok(result))
    .map((data) => data);
};
```

```
Type 'Err<unknown> | Ok<boolean>' is not assignable to type 'Result<boolean, Error>'.
  Type 'Err<unknown>' is not assignable to type 'Result<boolean, Error>'.
    Type 'ErrImpl<unknown>' is not assignable to type 'Err<Error>'.
      Type '{}' is missing the following properties from type 'Error': name, message

   return getStatus(true)
     ~~~~~~~~~~~~~~~~~~~~~~
     .andThen((result) => Ok(result))
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    .map((data) => data);
```

**Comment by vultix** [comment]
@datapain I'm not sure there's much we can do here.  Typescript 4.2 introduced breaking changes that broke this library.  In turn, the fixes for Typescript 4.2 that were released in ts-results 3.2.0 aren't compatible with earlier versions of typescript.

**Comment by datapain** [comment]
Okay, thank you, i update the message and i got another error:

```
import { Result, Ok, Err } from 'ts-results';

const getStatus = (payload: boolean): Result<boolean, Error> => {
  if (payload) {
    return Ok(payload);
  }
  return Err(new Error('Payload is false'));
};

const processStatus = (): Result<boolean, Error> => {
  return getStatus(true)
    .andThen((result) => Ok(result))
    .map((data) => data);
};

```
```
Type 'Err<unknown> | Ok<boolean>' is not assignable to type 'Result<boolean, Error>'.
  Type 'Err<unknown>' is not assignable to type 'Result<boolean, Error>'.
    Type 'ErrImpl<unknown>' is not assignable to type 'Err<Error>'.
      Type '{}' is missing the following properties from type 'Error': name, message

   return getStatus(true)
     ~~~~~~~~~~~~~~~~~~~~~~
     .andThen((result) => Ok(result))
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    .map((data) => data);
```
is this right to expect a Result when you return the map OK?

**Comment by vultix** [comment]
@datapain Good catch.  Fixed by 1ca8c75e736e182c8351b2e3922efc0012354ad3

**Comment by datapain** [comment]
Thank you.

Your fix was fixed the first error too
